schema/naming.go

```diff
@@ -32,6 +32,17 @@ type NamingStrategy struct {
 	SingularTable bool
 	NameReplacer  Replacer
 	NoLowerCase   bool
+	NamingStrategyConfig
+}
+
+// This struct is used to configure the behavior NamingStrategy, according to DB.
+// For example, in MySQL, the maximum length of an identifier is 64 characters.
+// In PostgreSQL, the maximum length of an identifier is 63 characters.
+// In SQL Server, the maximum length of an identifier is 128 characters.
+// In SQLite, the maximum length of an identifier is unlimited.
+// In future, we may add more options to NamingStrategyConfig.
+type NamingStrategyConfig struct {
+	IdentifierMaxLength int
```

Comment:
Don't need the `NamingStrategyConfig` struct? we can just move the `IdentifierMaxLength` field to `NamingStrategy`